### PR TITLE
Wait for RUNNING status

### PR DESCRIPTION
**Pull request by dagguh**
Wait for RUNNING /status

Sometimes the HTTP thread handling `waitForUpgrades` seems "stuck":
<details><summary>StuckThreadDetectionValve.notifyStuckThreadDetected</summary>
<p>
```
11-Jun-2024 10:46:56.661 WARNING [Catalina-utility-1] org.apache.catalina.valves.StuckThreadDetectionValve.notifyStuckThreadDetected Thread [http-nio-8080-exec-1] (id=[38]) has been active for [123,116] milliseconds (since [6/11/24 10:44 AM]) to serve the same request for [http://localhost:8080/rest/api/2/upgrade] and may be stuck (configured threshold for this StuckThreadDetectionValve is [120] seconds). There is/are [1] thread(s) in total that are monitored by this Valve and may be stuck.
	java.lang.Throwable
		at java.base@17.0.11/jdk.internal.misc.Unsafe.park(Native Method)
		at java.base@17.0.11/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
		at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:561)
		at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:111)
		at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:247)
		at com.google.common.cache.LocalCache$LoadingValueReference.waitForValue(LocalCache.java:3628)
		at com.google.common.cache.LocalCache$Segment.waitForLoadingValue(LocalCache.java:2211)
		at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2198)
		at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2081)
		at com.google.common.cache.LocalCache.get(LocalCache.java:4019)
		at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4042)
		at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5024)
		at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5031)
		at org.ofbiz.core.entity.GenericDelegator.getGenericDelegator(GenericDelegator.java:109)
		at com.atlassian.core.ofbiz.CoreFactory.getGenericDelegator(CoreFactory.java:39)
		at com.atlassian.jira.database.BootstrapQueryDslAccessor.createDelegate(BootstrapQueryDslAccessor.java:79)
		at com.atlassian.jira.database.BootstrapQueryDslAccessor.initDelegate(BootstrapQueryDslAccessor.java:71)
		at com.atlassian.jira.database.BootstrapQueryDslAccessor.getDelegate(BootstrapQueryDslAccessor.java:62)
		at com.atlassian.jira.database.BootstrapQueryDslAccessor.withNewConnection(BootstrapQueryDslAccessor.java:47)
		at com.atlassian.jira.config.properties.v2.OfBizPropertySetApplicationPropertiesDao.query(OfBizPropertySetApplicationPropertiesDao.java:360)
		at com.atlassian.jira.config.properties.v2.OfBizPropertySetApplicationPropertiesDao.exists(OfBizPropertySetApplicationPropertiesDao.java:118)
		at com.atlassian.jira.config.properties.v2.DefaultApplicationPropertiesManager.exists(DefaultApplicationPropertiesManager.java:34)
		at com.atlassian.jira.config.properties.v2.ApplicationPropertiesBackedPropertySet.exists(ApplicationPropertiesBackedPropertySet.java:90)
		at com.atlassian.jira.config.properties.ApplicationPropertiesStore.getOption(ApplicationPropertiesStore.java:166)
		at com.atlassian.jira.config.properties.ApplicationPropertiesImpl.getOption(ApplicationPropertiesImpl.java:93)
		at com.atlassian.jira.web.filters.gzip.JiraGzipFilter$JiraGzipFilterIntegration.lambda$useGzip$0(JiraGzipFilter.java:25)
		at com.atlassian.jira.web.filters.gzip.JiraGzipFilter$JiraGzipFilterIntegration$$Lambda$523/0x00007f3fd853cab0.apply(Unknown Source)
		at java.base@17.0.11/java.util.Optional.map(Optional.java:260)
		at com.atlassian.jira.web.filters.gzip.JiraGzipFilter$JiraGzipFilterIntegration.useGzip(JiraGzipFilter.java:23)
		at com.atlassian.gzipfilter.GzipFilter.doFilterInternal(GzipFilter.java:105)
		at com.atlassian.gzipfilter.GzipFilter.doFilter(GzipFilter.java:92)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
		at org.apache.catalina.valves.StuckThreadDetectionValve.invoke(StuckThreadDetectionValve.java:185)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:388)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1791)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base@17.0.11/java.lang.Thread.run(Thread.java:842)
```
</p>
</details> 

It takes a long time (over 2 minutes) and it might be benign, because 5 minutes later it's ok:
```
2024-06-11 10:51:50,186+0000 JIRA-Bootstrap INFO      [c.a.j.config.database.DatabaseConfigurationManagerImpl] The database is configured. Now running Database Checklist Launcher
```

But at the very least it generates log noise. Let's see if waiting for status before upgrades solves that.

### CODE REVIEW

**Comment by dagguh** [comment]
The other `catalina.out` did not have this warning. But it did take 8 minutes:
```
2024-06-06 13:14:57,958+0000 main INFO      [c.a.jira.startup.JiraStartupLogger] Running Jira startup checks.
2024-06-06 13:14:57,958+0000 main INFO      [c.a.jira.startup.JiraStartupLogger] Jira pre-database startup checks completed successfully.
2024-06-06 13:22:00,768+0000 main INFO      [c.a.j.config.database.DatabaseConfigurationManagerImpl] The database is configured. Now running Database Checklist Launcher
2024-06-06 13:22:00,770+0000 main INFO      [c.a.jira.startup.DatabaseChecklistLauncher] Jira database startup checks completed successfully.
```

So with the new /status polling we might get a faster startup. Or at very least, less WARN noise.
But I doubt it will fix the `Cannot invoke "org.apache.felix.framework.Felix.getBundleContext()" because "this.felix" is null`

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneStoppedNode.kt [hunk 1]
```diff
@@ -99,6 +103,32 @@ internal class StandaloneStoppedNode(
...
+        }
+        val statusEndpoint = URI("http://admin:$adminPasswordPlainText@localhost:8080/status")
+        pollJira(
+            launchTimeouts.initTimeout,
+            ssh,
+            threadDump,
+            "RUNNING state"
+        ) {
+            val response = ssh.safeExecute("curl $statusEndpoint", launchTimeouts.unresponsivenessTimeout).output
+            Json.createParser(StringReader(response)).use { jsonParser ->
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneStoppedNode.kt:122]: I had a choice between `jq` in Bash or `Json` in Java. Both had similar amount of _boilerplate_.

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneStoppedNode.kt [hunk 2]
```diff
@@ -99,6 +103,26 @@ internal class StandaloneStoppedNode(
...
 
+    private fun waitForRunning(
+        ssh: SshConnection,
+        threadDump: ThreadDump
+    ) {
+        if (!waitForRunning) {
+            logger.debug("Skipping Jira startup wait")
+            return
+        }
+        val statusEndpoint = URI("http://admin:$adminPasswordPlainText@localhost:8080/status")
```
Comment (mgrzaslewicz) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneStoppedNode.kt:114]: authorization is not needed for this endpoint

**Review by pczuj — APPROVED** [review]
LGTM

**Pull request merged**